uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveManagerTest.php

```diff
@@ -0,0 +1,124 @@
+
+  private $leaveManagerService;
+
+  private $loggedInContact;
+
+  public function setUp() {
+    $this->loggedInContact = ContactFabricator::fabricate();
+    $this->registerCurrentLoggedInContactInSession();
+    $this->leaveManagerService= new LeaveManagerService();
+  }
+
+  public function tearDown() {
+    $this->unregisterCurrentLoggedInContactFromSession();
+  }
+
+  public function testCurrentUserIsLeaveManagerOf() {
+    $staffMember = ContactFabricator::fabricate();
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+
+    $this->setContactAsLeaveApproverOf($this->loggedInContact, $staffMember);
+
+    $this->assertTrue($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+  }
+
+  public function testCurrentUserIsLeaveManagerOfWhenTheresNoActiveRelationshipForTheCurrentDate() {
+    $staffMember = ContactFabricator::fabricate();
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+
+    // Set a relationship in the past
+    $startDate = new DateTime('-10 days');
+    $endDate = new DateTime('-1 day');
+    $this->setContactAsLeaveApproverOf($this->loggedInContact, $staffMember, $startDate->format('Y-m-d'), $endDate->format('Y-m-d'));
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+
+    // Set a relationship in the future
+    $startDate = new DateTime('+1 day');
+    $this->setContactAsLeaveApproverOf($this->loggedInContact, $staffMember, $startDate->format('Y-m-d'));
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+  }
+
+  public function testCurrentUserIsLeaveManagerOfWhenTheCurrentRelationshipIsNotActive() {
+    $staffMember = ContactFabricator::fabricate();
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+    
+    $this->setContactAsLeaveApproverOf($this->loggedInContact, $staffMember, null, null, false);
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsLeaveManagerOf($staffMember['id']));
+  }
+
+  public function testCurrentUserIsAdmin() {
+    // First we need to make sure no permission is set. Note that if
+    // permissions is null, CRM_Core_Permission always returns true, so we must
+    // manually set to an empty array
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = [];
+
+    $this->assertFalse($this->leaveManagerService->currentUserIsAdmin());
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = ['administer leave and absences'];
+
+    $this->assertTrue($this->leaveManagerService->currentUserIsAdmin());
+  }
+
+  private function setContactAsLeaveApproverOf($leaveApprover, $contact, $startDate = null, $endDate = null, $isActive = true) {
+    $relationshipType = $this->getLeaveApproverRelationshipType();
+
+    civicrm_api3('Relationship', 'create', array(
```

Comment:
The Fabricator for RelationshipType already exists (you created it yourself!), but it has not been merged to la-milestone-3 yet. I would prefer not to use it for now and avoid a merge conflict once we merge staging into la-milestone-3.

As for the Relationship Fabricator, I agree we should create it, but would you think it's ok to do that on another PR? On PCHR-1798, I'm going to do some changes on this code and then I can add the fabricator. What do you think?